rclc_parameter/src/rclc_parameter/parameter_server.c

```diff
@@ -44,24 +52,43 @@ rclc_parameter_server_describe_service_callback(
   DescribeParameters_Request * request = (DescribeParameters_Request *) req;
   DescribeParameters_Response * response = (DescribeParameters_Response *) res;
 
-  size_t size = (request->names.size > param_server->parameter_list.size) ?
-    param_server->parameter_list.size :
-    request->names.size;
-
-  response->descriptors.size = size;
+  if (request->names.size > response->descriptors.capacity) {
+    response->descriptors.size = 0;
+    return;
+  }
 
-  for (size_t i = 0; i < size; i++) {
-    rclc_parameter_set_string(
-      &response->descriptors.data[i].name,
-      request->names.data[i].data);
+  response->descriptors.size = request->names.size;
 
-    Parameter * parameter = rclc_parameter_search(
+  for (size_t i = 0; i < request->names.size; i++) {
+    size_t index = rclc_parameter_search_index(
       &param_server->parameter_list,
       request->names.data[i].data);
 
-    response->descriptors.data[i].type = (parameter != NULL) ?
-      parameter->value.type :
-      RCLC_PARAMETER_NOT_SET;
+    ParameterDescriptor * response_descriptor = &response->descriptors.data[i];
+
+    // Reset response values
+    response_descriptor->type = RCLC_PARAMETER_NOT_SET;
+    response_descriptor->read_only = false;
+    response_descriptor->floating_point_range.size = 0;
+    response_descriptor->integer_range.size = 0;
```

Comment:
If `RCLC_PARAMETER_NOT_SET` is equal to `0` this can be:

```c
    response_descriptor = (ParameterDescriptor){0};
```

Reply:
Not really, as `(ParameterDescriptor){0};` will set string values as NULL pointer with capacity and size 0, but a empty string is expected to have this values:
```
data[0] = '\0'
size=0
capacity=1`
```

Otherwise, we get this error:
```
string capacity not greater than size
[ERROR] [1648720818.068012663] [rclc]: [rclc_execute] Error in rcl_send_response: cannot publish data, at /tmp/binarydeb/ros-galactic-rmw-fastrtps-shared-cpp-5.0.1/src/rmw_response.cpp:141, at /tmp/binarydeb/ros-galactic-rcl-3.1.2/src/rcl/service.c:287
```

Reply:
Ok with that